### PR TITLE
feat(source): use the default_provider for AWS-related connectors

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See background at #17851

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]

**Comment by yuhao-su** [comment]
#11086 We had an issue about this. Do we have a discussion context about why allowing this now?

**Comment by fuyufjh** [comment]
> #11086 We had an issue about this. Do we have a discussion context about why allowing this now?

Yes. See the background https://risingwave-labs.slack.com/archives/C03L8DNMDDW/p1722922448614259?thread_ts=1722281609.938729&cid=C03L8DNMDDW

> I feel we might need to review our decision before. Actually many of our (paid) customers tend to use BYOC or even on-prem. This seems to be a reasonable requirement in these cases.

File: src/connector/src/connector_common/common.rs [hunk 1]
```diff
@@ -46,6 +46,10 @@ pub const PRIVATE_LINK_TARGETS_KEY: &str = "privatelink.targets";
 
 const AWS_MSK_IAM_AUTH: &str = "AWS_MSK_IAM";
 
+/// The environment variable to disable using default credential from environment.
+/// It's recommended to set this variable to `false` in cloud hosting environment.
+const DISABLE_DEFAULT_CREDENTIAL: &str = "DISABLE_DEFAULT_CREDENTIAL";
```
Comment (arkbriar) [inline src/connector/src/connector_common/common.rs:51]: Shall it affect services from cloud vendors besides AWS as well?

Comment (fuyufjh) [reply src/connector/src/connector_common/common.rs:51]: If the idea looks good to you, I'll add the support of Azure/GCP in future PRs. This PR was supposed to complete #17851

Comment (arkbriar) [reply src/connector/src/connector_common/common.rs:51]: I see. Thanks!

**Review by arkbriar — APPROVED** [review]
LGTM!

**Pull request merged**